### DISCUSSION TITLE
& seems to be encoded?

### DISCUSSION THREAD

**Discussion by zykeee** [remarkjs/remark#1384] [Q&A] [answered]
When parsing from markdown to html like:
```js
import rehypeSanitize from "rehype-sanitize"
import rehypeStringify from "rehype-stringify"
import remarkParse from "remark-parse"
import remarkRehype from "remark-rehype"
import { unified } from "unified"

const md = "&"
const processed = unified()
	.use(remarkParse)
	.use(remarkRehype)
	.use(rehypeSanitize)
	.use(rehypeStringify)
	.processSync(md)
	.value.toString()

console.log(processed)
```

The result is `<p>&#x26;</p>`, seems to be encoded; but for others like `'`, works just fine.
**Is this an expected behavior and why if it is?**

<img width="808" alt="image" src="https://github.com/user-attachments/assets/f04fe0b9-5fd4-454a-b307-6bd930117405">

**Comment by wooorm** [comment, marked as answer]
Yes, the way things are, is expected.
There are many options, which you can use to configure things. 
See for example https://github.com/rehypejs/rehype/tree/main/packages/rehype-stringify

**Reply by zykeee** [reply]
Thanks for your reply.
I haven't found an option to change this behavior, what fits my scenario is to get `<p>&</p>` instead of `<p>&#x26;</p>`.
So what I do is to add `.replaceAll("&#x26;", "&")` after processing it.

**Reply by wooorm** [reply]
This is an XY problem.

You do not need to do that.

Please ask about what your actual goals are

**Reply by zykeee** [reply]
You are right.
I just notice that the gap here is browser would render `<p>&#x26;</p>` correctly as '&' but the lib I'm using would not. ([react-pdf-html](https://github.com/danomatic/react-pdf-html))
So when I pass the parsed html to it, it shows &#x26 like:
<img width="1097" alt="image" src="https://github.com/user-attachments/assets/3dfd41d3-a4a9-4680-85dc-e8b9993f0eae">

**Reply by wooorm** [reply]
Right, so that sounds like a bug there: if it accepts HTML, it should accept character references (which is the name for these things).

Your flow of things might not be optimal: you use these tools to generate a string of HTML, and then parse that HTML, and turn it into React.
I am sure there are tools that can instead turn markdown into PDFs directly, or through React, but without the serialized HTML in between!
